src/qibosoq/rfsoc_server.py

```diff
@@ -16,12 +16,76 @@
 
 import qibosoq.configuration as cfg
 from qibosoq.components import Config, OperationCode, Pulse, Qubit, Sweeper
-from qibosoq.qick_programs import ExecutePulseSequence, ExecuteSweeps
+from qibosoq.programs.pulse_sequence import ExecutePulseSequence
+from qibosoq.programs.sweepers import ExecuteSweeps
 
 logger = logging.getLogger(cfg.MAIN_LOGGER_NAME)
 qick_logger = logging.getLogger(cfg.PROGRAM_LOGGER_NAME)
 
 
+def execute_program(self, data: dict) -> dict:
```

Comment:
Maybe don't call it `self` :)

(it's not a special/reserved word for Python, but it is a bit special for people reading Python)

Reply:
Moreover, they are now swapped in the call, so you should either swap here:
```suggestion
def execute_program(data: dict, soc) -> dict:
```
or make the call consistent.

Reply:
I thought I did it, but I didn't... thanks!